vnext/Microsoft.ReactNative.Cxx/JSI/JsiAbiApi.cpp

```diff
@@ -634,9 +634,23 @@ bool JsiAbiRuntime::instanceOf(const Object &o, const Function &f) try {
   throw;
 }
 
+template <typename T>
+struct AutoRestore {
+  AutoRestore(T *var, T value) : m_var{var}, m_value{std::exchange(*var, value)} {}
+
+  ~AutoRestore() {
+    *m_var = m_value;
+  }
+
+ private:
+  T *m_var;
+  T m_value;
+};
```

Comment:
consider putting this in a common header if you're going to use it from more than one place